holoviews/core/options.py

```diff
@@ -612,6 +613,7 @@ def __init__(self, key=None, allowed_keywords=[], merge_keywords=True, max_cycle
                             else Keywords(allowed_keywords))
         super(Options, self).__init__(allowed_keywords=allowed_keywords,
                                       merge_keywords=merge_keywords, key=key)
+        self.backend = backend
```

Comment:
I can't see all the logic involved here, but I think this should be something like `self.backend = backend if backend is not None else current_backend`.  That way the backend in use when these objects are created is bound to them so that it's clear which backend these options apply to.

Reply:
Hmm, you can define these options backends before a backend is loaded and then apply them when it is loaded so it's not possible to rely on this being defined. Therefore I don't think it should be bound to the object here.

Reply:
What will `Store.current_backend` be if there is no backend loaded?  Won't it (correctly) be None?

Reply:
>What will `Store.current_backend` be if there is no backend loaded?

This actually needs looking at since at least under some conditions it's 'matplotlib' by default. 

However even if it was consistently None (which it should be), this would lead to unpredictable behavior. Options defined before a backend is loaded would default to whatever backend is used when applying the options while if a backend is loaded it immediately defaults to that. Seems like a recipe for confusion to me. I think if unspecified it should therefore default to whatever it is when it is applied, because at that point a backend has to be loaded.

Reply:
> under some conditions it's 'matplotlib' by default.

Definitely need to get rid of that!  I would hope people would be seeing validation errors if they start defining Bokeh options while the backend is defaulting to Matplotlib, so hopefully this would be a weird situation to get into in practice.

All of our examples consistently show loading an extension before doing anything, so I don't think we should be torturing the design to support people who don't do that.

Reply:
I agree with @philippjfr on all counts here and that there should be no more historical 'matplotlib' by default (which I *hope* is only when you do `hv.extension()`).

> All of our examples consistently show loading an extension before doing anything, so I don't think we should be torturing the design to support people who don't do that.

I completely disagree. One of the goals of holoviews (at least for me) is to declaratively build objects in a very bare environment (e.g on an HPC cluster) that you can then export and load into a richer environment that does have plotting dependencies. Requiring backends to always exist first basically would break the boundary between holoviews-core and the plotting extensions.